powermon/ports/__init__.py

```diff
@@ -1,42 +1,66 @@
+""" ports/__init__.py """
 import logging
+from enum import auto
+
+from strenum import LowercaseStrEnum
 
-from powermon.ports.abstractport import PortType
 from powermon.ports.serialport import SerialPort
+from powermon.ports.testport import TestPort
 from powermon.ports.usbport import USBPort
 
 # Set-up logger
 log = logging.getLogger("ports")
 
 
+class PortType(LowercaseStrEnum):
+    """ enumeration of supported / known port types """
+    UNKNOWN = auto()
+    TEST = auto()
+    SERIAL = auto()
+    USB = auto()
+
+    BLE = auto()
+
+    JKBLE = auto()
+    MQTT = auto()
+    VSERIAL = auto()
+    DALYSERIAL = auto()
+    ESP32 = auto()
+
+
 def getPortFromConfig(port_config):
-    log.debug(f"port_config: {port_config}")
+    """ get a port object from config data """
+    log.debug("port_config: %s", port_config)
 
-    portObject = None
+    port_object = None
     if not port_config:
-        log.info("no port config supplied, defaulting to test port")
+        log.info("no port config supplied, defaulting to test port")  # QUESTION: does this make sense, maybe should return None
         port_config = {"type": "test", "protocol": "PI30"}
 
     # port type is mandatory
-    portType = port_config.get("type")
-    log.debug(f"portType: {portType}")
+    port_type = port_config.get("type")
+    log.debug("portType: %s", port_type)
 
     # return None if port type is not defined
-    if portType is None:
+    if port_type is None:
         return None
 
     # build port object
-    if portType == PortType.SERIAL:
-        portObject = SerialPort.fromConfig(config=port_config)
-    elif portType == PortType.USB:
-        portObject = USBPort.fromConfig(config=port_config)
+    match port_type:
```

Comment:
I'm adopting the match pattern in preference to if, elif, else blocks where possible